core/test/variable_test.cpp

```diff
@@ -958,3 +958,11 @@ TEST(VariableTest, construct_proxy_dims) {
   Variable vv(var.slice({Dim::X, 0, 2}));
   ASSERT_NO_THROW(Variable(var.slice({Dim::X, 0, 2}), Dimensions(Dim::Y, 2)));
 }
+
+TEST(VariableTest, construct_mult_dev_unit) {
+  Variable refDiv = makeVariable<float>(
+      {}, units::Unit(units::dimensionless) / units::Unit(units::m), {1.0f});
+  Variable refMult = makeVariable<int32_t>({}, units::Unit(units::kg), {1});
+  EXPECT_EQ(1.0f / units::Unit(units::m), refDiv);
+  EXPECT_EQ(int32_t(1) * units::Unit(units::kg), refMult);
```

Comment:
No it doesn't because `units::kg` has some boost type which has it's own `*` / `/` operators, which doesn't produce any `Variable`. @SimonHeybrock, do you have any idea how to work around this?

Reply:
Right. Maybe we can consider changing `units::kg` and the others to be a (`constexpr`) `Unit` instead? But we should analyze the pros and cons in more detail and not as part of the PR.

Reply:
I agree, in general, I'm not happy with current units at all, we should discuss.